### PR TITLE
Implements #537 KElbowVisualizer can accept an iterable of integers.

### PR DESCRIPTION
**Pull request by drwaterman**
Functionality is unchanged if k is a single int or a tuple of 2 ints. New functionality allows the user to pass an iterable of discrete ints to use as the k values.

### CODE REVIEW

**Comment by lwgray** [comment]
@drwaterman thanks for the PR/Contribution  I will reveiw it ASAP

File: tests/test_cluster/test_elbow.py [hunk 1]
```diff
@@ -184,11 +184,31 @@ def test_invalid_k(self):
...
+        visualizer = KElbowVisualizer(KMeans(), k=(4, 12))
+        assert visualizer.k_values_ == list(np.arange(4, 12))
+
+        visualizer = KElbowVisualizer(KMeans(), k=np.arange(10, 100, 10))
+        assert visualizer.k_values_ == list(np.arange(10, 100, 10))
+
+        visualizer = KElbowVisualizer(KMeans(),
+                                      k=[10, 20, 30, 40, 50, 60, 70, 80, 90])
+        assert visualizer.k_values_ == list(np.arange(10, 100, 10))
+
```
Comment (lwgray) [inline tests/test_cluster/test_elbow.py]: - Great tests... I believe you covered all inputs types.  
- I'm wondering if you might want to test that a `YellowbrickValueError` is raised when non-integer values are input into `k`.  @bbengfort and @rebeccabilbro can you chime in and let us know if this extra test is necessary?

File: yellowbrick/cluster/elbow.py [hunk 2]
```diff
@@ -150,10 +151,11 @@ class KElbowVisualizer(ClusteringScoreVisualizer):
...
 
-    k : integer or tuple
-        The range of k to compute silhouette scores for. If a single integer
-        is specified, then will compute the range (2,k) otherwise the
-        specified range in the tuple is used.
+    k : integer, tuple, or iterable
+        The k values to compute silhouette scores for. If a single integer
+        is specified, then will compute the range (2,k). If a tuple of 2
+        integers is specified, then k will be in np.arange(k[0], k[1]).
+        Otherwise, specify an iterable of integers to use as values for k.
```
Comment (lwgray) [inline yellowbrick/cluster/elbow.py]: Thanks for updating this doc

File: yellowbrick/cluster/elbow.py [hunk 3]
```diff
@@ -220,19 +222,19 @@ def __init__(self, model, ax=None, k=10,
...
-        except:
+        elif isinstance(k, collections.Iterable) and \
+                all(isinstance(x, (int, np.integer)) for x in k):
+            self.k_values_ = list(k)
+        else:
             raise YellowbrickValueError((
-                "Specify a range or maximal K value, the value '{}' "
-                "is not a valid argument for K.".format(k)
+                "Specify an iterable of integers, a range, or maximal K value,"
+                " the value '{}' is not a valid argument for K.".format(k)
```
Comment (lwgray) [inline yellowbrick/cluster/elbow.py]: - This looks good...
- great use of `all` to make sure only integers are present.
- Thanks for updating the Error messages

File: tests/test_cluster/test_elbow.py (hunk 1 above)
Comment (rebeccabilbro) [reply tests/test_cluster/test_elbow.py]: Hey there @lwgray - by non-integer are you thinking floats? Looks like @drwaterman already has test cases for strings.

**Pull request merged**